js/main.js

```diff
@@ -70,31 +70,35 @@ function onDOMLoad(event) {
     var entryRendered = renderEntry(data.entries[entryNum]);
     $entryList.appendChild(entryRendered);
   }
-  if (data.view === 'entries') {
-    $formDiv.className = data.view + ' hidden';
-    data.view = 'entries';
-    $entriesDiv.className = data.view;
+  for (var i = 0; i < $views.length; i++) {
+    if ($views[i].getAttribute('data-view') === data.view) {
+      $views[i].className = 'view';
+    } else {
+      $views[i].className = 'view hidden';
+    }
   }
 }
 
 window.addEventListener('DOMContentLoaded', onDOMLoad);
 
-var $entriesTab = document.querySelector('.entriesTab');
 var $entriesDiv = document.querySelector('.entries');
-var $newButton = document.querySelector('.newButton');
 var $formDiv = document.querySelector('.entry-form');
 
-function onNewClick(event) {
-  $entriesDiv.className = data.view + ' hidden';
-  data.view = 'entry-form';
-  $formDiv.className = data.view;
-}
-
-function onEntriesClick(event) {
-  $formDiv.className = data.view + ' hidden';
-  data.view = 'entries';
-  $entriesDiv.className = data.view;
+var $newButton = document.querySelector('.newButton');
+var $entriesTab = document.querySelector('.entriesTab');
+var $views = document.querySelectorAll('.view');
+
+function switchViews(event) {
+  var targetDataView = event.target.getAttribute('data-view');
+  for (var i = 0; i < $views.length; i++) {
+    if ($views[i].getAttribute('data-view') === targetDataView) {
+      $views[i].className = 'view';
+      data.view = targetDataView;
+    } else {
+      $views[i].className = 'view hidden';
+    }
+  }
```

Comment:
This is the sort of thing I had in mind, but you shouldn't make this function reliant on an `event` object. You want this to have a parameter named something like `entry` which is a string of the name of the view to go to. That way, you can use this function to switch between views in any context, not just an event handler context 🤓